lib/autoupdate/tools/helper.go

```diff
@@ -37,37 +40,69 @@ var (
 	version = teleport.Version
 	// baseURL is CDN URL for downloading official Teleport packages.
 	baseURL = autoupdate.DefaultBaseURL
+	// ErrDisabled returns when home folder isn't set
+	ErrDisabled = errors.New("client tools update is disabled")
 )
 
-// CheckAndUpdateLocal verifies if the TELEPORT_TOOLS_VERSION environment variable
-// is set and a version is defined (or disabled by setting it to "off"). The requested
-// version is compared with the current client tools version. If they differ, the version
-// package is downloaded, extracted to the client tools directory, and re-executed
-// with the updated version.
-// If $TELEPORT_HOME/bin contains downloaded client tools, it always re-executes
-// using the version from the home directory.
-func CheckAndUpdateLocal(ctx context.Context, reExecArgs []string) error {
+// newUpdater inits the updater with default base ULR and tools directory
+// from Teleport home directory.
+func newUpdater() (*Updater, error) {
 	toolsDir, err := Dir()
 	if err != nil {
-		slog.WarnContext(ctx, "Client tools update is disabled", "error", err)
-		return nil
+		return nil, ErrDisabled
```

Comment:
Can we keep the original error or at least log a warning? Here if something goes wrong when looking up the user's home dir we will swallow the original error and say "tools updates are disabled". This can lead to hard to troubleshoot bug.